### ISSUE TITLE
Why are plots for the full training dataset and the test set so different from the replica

### ISSUE THREAD

**Issue by gordonwatts** [gordonwatts/CalRatioTrainer#178]
This is scaring me. Here is `llp_pt` for felix's test and main training data:

![image](https://github.com/gordonwatts/CalRatioTrainer/assets/1778366/428ae5bf-8634-4fd1-b854-ed8aac66234f)

And for Fexlix's test dataset and the replica dataset:

![image](https://github.com/gordonwatts/CalRatioTrainer/assets/1778366/293880b8-3aff-4233-85ba-b7f2d89d8034)

This points to a plotting bug - but until we track that don't I don't know we can actually trust any other plots!

**Comment by gordonwatts** [comment]
The fact that is such a thin spike and the limits go up to 100 rather than a fraction is probably a big clue as to what is going on here.

**Comment by gordonwatts** [comment]
Note there is no spike at zero for the `llp_pt` when we look at both of the training files. In short- something has gone wrong here with the plotting code. That set of plots looks a lot better than the one that compares my training data to Felix's.

**Comment by gordonwatts** [comment]
Ok - the problem might be the labeling!!

and here is some code from `pre_process.py`:

```python
    len_data = data[data['label'] == 2].shape[0]
    len_qcd = data[data['label'] == 0].shape[0]
    qcd_ratio = len_qcd/len_data
    temp_qcd = data[data['label'] == 0]
    temp_qcd = temp_qcd.sample(1/qcd_ratio)
    data = (data[data['label'] == 2]).merge(temp_qcd)
    data = data.sample(1)
```

And I found this in the original `process_events.py`:

```python
    signal_arrays = arrays[arrays.signal == 1]
    QCD_arrays = arrays[arrays.QCD == 1]
    BIB_arrays = arrays[arrays.BIB == 1]
```

Where `arrays` was read via `uproot` form our raw file. So, I can see those getting imported, but I can't see how label is getting imported or set! Crazy!

Alex's code has signal as label zero, BIB as two, and QCD as one.

And here is in `utils.py` and `low_or_high_pt_selection`:

```python
        signal = df.loc[ ((df['label'] == 1) & (df['llp_mH'] == 60)) | ((df['label'] == 1) & (df['llp_mH'] == 125)) | ((df['label'] == 1) & (df['llp_mH'] == 200))]
        qcd = df.loc[ ((df['label'] == 0) & (df['llp_mH'] == 60)) | ((df['label'] == 0) & (df['llp_mH'] == 125)) | ((df['label'] == 0) & (df['llp_mH'] == 200))]
        bib = df.loc[ ((df['label'] == 2) & (df['llp_mH'] == 60)) | ((df['label'] == 2) & (df['llp_mH'] == 125)) | ((df['label'] == 2) & (df['llp_mH'] == 200))]
```

And here is what we are expecting:

```
| `main_training_file` | This training file contains the three types of jets the NN is to discriminate against - ones from decays in the calorimeter, ones from QCD, and ones from BIB (labeled with 0, 1, or 2). |
```

**Comment by gordonwatts** [comment]
So, proper is, in a training file:

* signal == 1
* QCD == 0
* BIB == 1

And for an adversary training file:

**Comment by gordonwatts** [comment]
The labeling will be fixed by #179. Hopefully when that is done this will be fixed as well.

**Comment by gordonwatts** [comment]
Let's do a rundown now:

* [x] Multijet matches
* [x] Signal
    * Other than and `track_pt_0`, `track_eta_1`, this looks quite good. That problem "feels" like it is due to another problem, however.
* [x] BIB
    * Same issues as in Signal.

While the track stuff seems to have a problem, this isn't labeling. In particular labeling issue. So we can close this, and open a new one for this other error.
